github/git_commits.go

```diff
@@ -88,29 +88,33 @@ type createCommit struct {
 }
 
 // CreateCommit creates a new commit in a repository.
+// commit must not be nil.
 //
 // The commit.Committer is optional and will be filled with the commit.Author
 // data if omitted. If the commit.Author is omitted, it will be filled in with
 // the authenticated user’s information and the current date.
 //
 // GitHub API docs: https://developer.github.com/v3/git/commits/#create-a-commit
 func (s *GitService) CreateCommit(ctx context.Context, owner string, repo string, commit *Commit) (*Commit, *Response, error) {
+	if commit == nil {
+		return nil, nil, fmt.Errorf("commit must be provided")
```

Comment:
[Effective Go](https://golang.org/doc/effective_go.html#panic) recommends:

> ... real library functions should avoid panic.

and I strongly agree. I would much rather that my server/program spit out an error message and continue on (or choose to die, depending on its particular circumstances) than to have some library choose for me to kill my program.